docs/core/deploying/native-aot/interop.md

```diff
@@ -61,17 +61,23 @@ Examples:
 </ItemGroup>
 ```
 
-To specify additional flags to the native linker, use the `<LinkerArg>` item.
+To specify additional dependencies and flags to the native linker, use the `<NativeSystemLibrary>`, `<NativeFramework>` and `<ExtraLinkerArg>`.
 
 Examples:
 
 ```xml
 <ItemGroup>
   <!-- link.exe is used as the linker on Windows -->
-  <LinkerArg Include="/DEPENDENTLOADFLAG:0x800" Condition="$(RuntimeIdentifier.StartsWith('win'))" />
+  <ExtraLinkerArg Include="/DEPENDENTLOADFLAG:0x800" Condition="$(RuntimeIdentifier.StartsWith('win'))" />
```

Comment:
`ExtraLinkerArg` and `NativeSystemLibrary` are Unix-specific. This change does not work on Windows.

These options have been internal implementation detail, not meant for general use.

Reply:
> These options have been internal implementation detail, not meant for general use.

This needs some clarification. https://github.com/dotnet/runtime/issues/97414#issuecomment-1907398620 says the exact opposite - i.e. that `LinkerArg` is the implementation detail

Reply:
That's unfortunate. `LinkerArg` was considered to be too hard to adapt for the use in macOS/iOS workload, so the decision was to use specific properties to transfer the linker args in an efficient way that doesn't require complicated parsing.

(`LinkerArg` can contain spaces and `-Wl,` prefixes and it was overall hard to consume in the workload step; ref: https://github.com/xamarin/xamarin-macios/issues/18819)

Reply:
We can work on this to make it better for .NET 9. Documenting the current state may be confusing.

Reply:
The issue is that what the current documentation describes just doesn't work at all on net8.0-mac. with no indication whatsoever that would be the case. With the incorrect docs and no knowledge of the internals, all I could do was create the issue (https://github.com/dotnet/runtime/issues/97414) and that's how I was able to find out it needs to be implemented either with `ExtraLinkerArg` or NativeSystemLibrary`, not what the docs say.

Changing the behaviour for .NET 9 doesn't help anyone trying to use this now so from my POV, docs need fixing, one way or another.

Reply:
I think Xamarin needs to document how to pass linker args using their mechanism. The native AOT .targets (that this page documents) are not involved in executing the linker for Xamarin use cases because Xamarin hijacks this part of the build.

The format of arguments is different when Xamarin is involved, like Filip said in https://github.com/dotnet/docs/pull/39230#discussion_r1465051340. These are not exchangeable with `LinkerArg` and Xamarin has its own rules about how the argument string should look like. `ExtraLinkerArg` will likely just run into the same issues with spaces and escaping that were the original motivation for why Xamarin stopped looking at `LinkerArg`. It will only work sometimes, for some arguments. Xamarin needs to document how to pass extra flags along with expected escaping rules. Our escaping rules that we document are "this is the command line to clang/gcc", but that's not the case for Xamarin targets.

> We can work on this to make it better for .NET 9. Documenting the current state may be confusing.

Agreed - the Xamarin targets with native AOT are documented to be experimental in .NET 8. Not everything works. This will be one of those things.

Reply:
@rolfbjarne I suggest we add a section in https://github.com/xamarin/xamarin-macios/blob/main/docs/nativeaot.md about extending linker options in this scenario.
What do you think?

Reply:
`ExtraLinkerArg` should have the same quoting/spacing behavior we expect (https://github.com/dotnet/runtime/pull/88294#issuecomment-1663268829), so we could either document that item group (which would work in .NET 8), or come up with a new item group for public consumption (which wouldn't work until .NET 9).

Reply:
This is getting out of hand so I'll leave the docs update to you, since I'd be just guessing which bits you'd want documented and where.

I'm closing the PR but feel free to reuse any of its content elsewhere.

Reply:
Thank you very much for your efforts on this topic. 
As we continuing our work on supporting NativeAOT on apple platforms, we will also invest in improving the documentation and overall user experience.
I assigned the originally reported issue to me and will make sure we fix it in the right place.